src/Clash/Cores/ClashFloPoCo/GenTemDSL.hs

```diff
@@ -0,0 +1,1306 @@
+--  > {-# ANN plusFloat (flopocoPrim 'plusFloat 'plusFloatBBF) #-}
+--  is equavilent to
+--
+--  > {-# ANN plusFloat (
+--  >    let
+--  >      primName = show 'plusFloat
+--  >      tfName = show 'plusFloatBBF
+--  >   in
+--  >      InlineYamlPrimitive [minBound..] [__i|
+--  >        BlackBoxHaskell:
+--  >          name: #{primName}
+--  >          templateFunction: #{tfName}
+--  >          workInfo: Always
+--  >      |]) #-}
+flopocoPrim ::
+  -- | Name of the Clash function corresponding to entity(VHDL) or module(Verilog, SystemVerilog)
+  Name ->
+  -- | Name of BlackBox function of the Clash function
+  Name ->
+  Primitive
+flopocoPrim fprimName ftfName =
+  let primName = fprimName
+      tfName = ftfName
+   in InlineYamlPrimitive
+        [minBound ..]
+        [__i|
+        BlackBoxHaskell:
+          name: #{primName}
+          templateFunction: #{tfName}
+          workInfo: Always
+        |]
+
+-- | Function to convert Maybe Int to Int
+convertMaybeToInt :: Maybe Int -> Int
+convertMaybeToInt = fromMaybe 0
+
+-- | Function to generate a black box function with dynamic names
+--
+--  Please see the source file if cannot see the content of the example
+--
+--  ==== __Example:__
+
+-- |
+-- @
+-- infoen = InfoEntity {name = Just "plusFloat", freq = Nothing, pipedep = Nothing, insig = Nothing, outsig = Nothing}
+-- \$(genBlackBoxFunction infoen)
+-- @
+--
+-- will generate this code during compile time
+--
+-- @
+-- plusFloatBBF :: BlackBoxFunction
+--    plusFloatBBF _ _ _ _
+--       = pure
+--          (Right
+--             (emptyBlackBoxMeta {bbKind = TDecl},
+--             BBFunction "plusFloatTF" 0 (plusFloatTF "plusFloat")))
+-- @
+genBlackBoxFunction :: InfoEntity -> Q [Dec]
+genBlackBoxFunction infoen = do
+  let entityName = fromMaybe "" (name infoen)
+  let tfName = mkName (entityName <> "TF")
+  let bbfName = mkName (entityName <> "BBF")
+  -- Create the function declaration
+  let funcDec =
+        FunD
+          bbfName
+          [ Clause
+              [WildP, WildP, WildP, WildP]
+              ( NormalB
+                  ( AppE
+                      (VarE 'pure)
+                      ( AppE
+                          (ConE 'Right)
+                          ( TupE
+                              [ Just (RecUpdE (VarE 'emptyBlackBoxMeta) [('bbKind, ConE 'TDecl)]),
+                                Just
+                                  ( AppE
+                                      ( AppE
+                                          (AppE (ConE 'BBFunction) (LitE (StringL (entityName <> "TF")))) -- 1st arg
+                                          (LitE (IntegerL 0)) -- 2nd arg
+                                      )
+                                      (AppE (VarE tfName) (LitE (StringL entityName))) -- 3rd arg
+                                  )
+                              ]
+                          )
+                      )
+                  )
+              )
+              []
+          ]

```

Comment:
This could be simplified to:
```haskell
  -- Create the function declaration
  funExpr <- [|pure (Right (emptyBlackBoxMeta {bbKind = TDecl}, BBFunction (entityName <> "TF") 0 ($(varE tfName) entityName)))|]
  let funcDec = FunD bbfName [Clause [WildP,WildP,WildP, WildP] (NormalB funExpr) []]
```